### PR TITLE
FTS - Can use 'similarity' ranking system

### PR DESCRIPTION
**Pull request by ger-benjamin**
@sbrunner How do you think I should adapt the tests ?

Note:
Later, I'll port this on master, but similarity will be the default system.

### CODE REVIEW

**Comment by kalbermattenm** [comment]
Cool ! Thanks a lot for that @ger-benjamin !

File: c2cgeoportal/tests/functional/test_fulltextsearch.py [hunk 1]
```diff
@@ -128,6 +148,12 @@ def tearDown():  # noqa
             FullTextSearch.label == "label5").delete()
         DBSession.query(FullTextSearch).filter(
             FullTextSearch.label == "label6").delete()
+        DBSession.query(FullTextSearch).filter(
+            FullTextSearch.label == "label7").delete()
+        DBSession.query(FullTextSearch).filter(
+            FullTextSearch.label == "label70").delete()
+        DBSession.query(FullTextSearch).filter(
+            FullTextSearch.label == "label71").delete()
```
Comment (sbrunner) [inline c2cgeoportal/tests/functional/test_fulltextsearch.py]: 71
70
to be sure that they are also ordered :-)

Comment (ger-benjamin) [reply c2cgeoportal/tests/functional/test_fulltextsearch.py]: You're right :-)

**Comment by ger-benjamin** [comment]
@yjacolin or @jwkaltz Can you take a look at my doc ? Thanks !

File: doc/integrator/fulltext_search.rst.mako [hunk 2]
```diff
@@ -209,6 +209,26 @@ following variables:
    default is 30.
 *  ``fulltextsearch_maxlimit`` the max possible limit, default is 200.
 
+Ranking system
+--------------
+
+By default, the fill text search uses `tsvector` and `ts_rank_cd` to rank your search results (See:
```
Comment (yjacolin) [inline doc/integrator/fulltext_search.rst.mako]: fill -> full

File: doc/integrator/fulltext_search.rst.mako [hunk 3]
```diff
@@ -209,6 +209,26 @@ following variables:
...
+`textsearch-controls <https://www.postgresql.org/docs/9.0/static/textsearch-controls.html>`_. These methods
+are useful to handle language-based strings. That means for instance that plural nouns are the same as
+singular nouns. This system only check if your search word exists in the result. That means that if you search
```
Comment (yjacolin) [inline doc/integrator/fulltext_search.rst.mako]: checkS

File: doc/integrator/fulltext_search.rst.mako [hunk 4]
```diff
@@ -209,6 +209,26 @@ following variables:
...
+`B 12 Zug`, `B 120 Zug` has the same weight because the system only see that the `12` exists in each case.
+
+Alternatively you can use the `similarity` system of the
+`pg_trgm module <https://www.postgresql.org/docs/9.0/static/pgtrgm.html>`. This is based only on the
+similarities of words, without language analysis, and it cares only about how near is your search with the
+result. `12` is nearer to `12` than `120`.
+To use this system, your request must contains the parameter `rank_system=similarity` and your database must
```
Comment (yjacolin) [inline doc/integrator/fulltext_search.rst.mako]: How to add such parameter in the FTS request? Do this need some configuration at the client side?

Comment (ger-benjamin) [reply doc/integrator/fulltext_search.rst.mako]: Yes, and the documentation for that will be on the client side.

File: doc/integrator/fulltext_search.rst.mako (hunk 2 above)
Comment (sbrunner) [reply doc/integrator/fulltext_search.rst.mako]: full-text search

**Comment by ger-benjamin** [comment]
@yjacolin done.
@sbrunner I still have this error: 
```
c2cgeoportal_1  | ProgrammingError: (psycopg2.ProgrammingError) function similarity(character varying, unknown) does not exist
c2cgeoportal_1  | LINE 3: ...ang IS NULL OR main.tsearch.lang = 'fr') ORDER BY similarity...
```

I don't know where to do my 
`CREATE EXTENSION IF NOT EXISTS pg_trgm;`

**Comment by sbrunner** [comment]
The extension is missing here: https://github.com/camptocamp/c2cgeoportal/blob/2.2/docker/test-db/10_init.sql

**Comment by sbrunner** [comment]
@yjacolin is it ok for you now?

**Pull request merged**